packages/eslint-plugin-office-addins/src/rules/call-sync-before-read.ts

```diff
@@ -30,6 +34,8 @@ export = {
       switch (topExpression.parent?.type) {
         case TSESTree.AST_NODE_TYPES.AssignmentExpression:
           return topExpression.parent.right === topExpression;
+        case TSESTree.AST_NODE_TYPES.CallExpression:
```

Comment:
I thought CallExpression itself indicated a method . . . why do we also need to check again?

Reply:
We didn't check for callExpression anywhere else... This is the first check for it because we want to know I the thing after the `.` is a property or a method.
If you are wondering why do we need to check with the `isProperty(node)`, it is because there are two types of methods:
`console.log(range.address)` will have a node type = callExpression as with `range.delete()`. So we need to somehow differentiate those two methods, as both have CallExpression on them.

The way I found was by checking with the `isProperty` function, which now I realize it should be renamed

Reply:
Why does it matter if a method call has an argument or not?